File: laoo/lib/core/master/master_group_codes.dart
/// Central definitions for master groups stored in dbo.TDSTMaster.
abstract final class MasterGroupCodes {
  static const province = '001';
  static const unit = '002';
  static const customerBusiness = '003';
  static const priceLevel = '004';
  static const customerGroup = '005';
  static const itemGroup = '006';
  static const itemType = '007';

  static const variableNames = <String, String>{
    province: 'MsProv',
    unit: 'MsUnit',
    customerBusiness: 'MsBusiness',
    priceLevel: 'MsPriceLevel',
    customerGroup: 'MsCusGroup',
    itemGroup: 'MsItemGroup',
    itemType: 'MsItemType',
  };

  static String? variableNameOf(String code) => variableNames[code];
}

File: projects/meeting/packages/dotnet/Laoo.Meeting.Module/Models/MasterGroupCodes.cs
namespace LaooMeetingApi.Models;

/// Central definitions for master groups stored in dbo.TDSTMaster.
public static class MasterGroupCodes
{
    public const string Province = "001";
    public const string Unit = "002";
    public const string CustomerBusiness = "003";
    public const string PriceLevel = "004";
    public const string CustomerGroup = "005";
    public const string ItemGroup = "006";
    public const string ItemType = "007";
    public const string FoodType = "011";

    public static readonly IReadOnlyDictionary<string, string> VariableNames =
        new Dictionary<string, string>(StringComparer.Ordinal)
        {
            [Province] = "MsProv",
            [Unit] = "MsUnit",
            [CustomerBusiness] = "MsBusiness",
            [PriceLevel] = "MsPriceLevel",
            [CustomerGroup] = "MsCusGroup",
            [ItemGroup] = "MsItemGroup",
            [ItemType] = "MsItemType",
            [FoodType] = "MsFoodType",
        };
}
